argusclient/client.py

```diff
@@ -591,6 +591,72 @@ def get_alerts_allinfo(self, ownerName=None, alertNameContains=None, shared=Fals
         """
         return self.argus._request("get", "alerts/allinfo", params=dict(ownername=ownerName, alertNameContains=alertNameContains, shared=shared, limit=limit))
 
+    '''
+    Functions for enabling composite alerts
+    '''
+
+    def get_composite_alert_children_info(self, alert_id):
+        if not alert_id: raise ValueError("Need to specify alert id")
+        uri_path = "alerts/{}/children/info".format(alert_id)
+        return self.argus._request("get", uri_path)
+
+    def create_composite_alert(self, payload):
+        if not payload: raise ValueError("Need to specify an payload")
+        alert_obj = self.argus._request("post", "alerts", dataObj=payload)
+        self._coll[alert_obj.id] = alert_obj
+        return alert_obj
+
+    def add_child_alert_to_composite_alert(self, alert_id, payload):
+        if not alert_id: raise ValueError("Need to specify composite alert id")
+        if not payload: raise ValueError("Need to specify  payload")
+        uri_path = "alerts/{}/children".format(alert_id)
+        alert_obj = self.argus._request("post", uri_path, dataObj=payload)
+        self._coll[alert_obj.id] = alert_obj
+        return alert_obj
+
+    def update_composite_alert(self, alert_id, payload):
+        if not alert_id: raise ValueError("Need to specify composite alert id")
+        if not payload: raise ValueError("Need to specify  payload")
+        uri_path = "alerts/{}".format(alert_id)
+        return self.argus._request("put", uri_path, dataObj=payload)
+
+    def delete_child_alert_from_composite_alert(self, comp_alert_id, child_alert_id):
+        """
+        Delete a child alert from a composite alert
+        """
+        if not comp_alert_id: raise ValueError("Need to specify composite alertid")
+        if not child_alert_id: raise ValueError("Need to specify a child alertid")
+        uri_path = "alerts/{}/children/{}".format(comp_alert_id, child_alert_id)
+        if id in self._coll:
+            del self._coll[child_alert_id]
```

Comment:
Does this operation only delete the association from the compose alert or does it also delete the alert? Isn't the alert with `child_alert_id` is still valid?

Reply:
Is there an action item on this?

Reply:
Yes its a typo it should be child_alert_id.  Thanks for catching it

Reply:
I see that you addressed the typo, but it is in the previous line. My question is still valid for this line, I am still not sure if you are just deleting the association only or even the alert. Don't you need another delete call for the `child_alert_id`? Or is that up to the client since the creation is also done by the client? The child alert objects would become garbage and leak otherwise.

Reply:
This is still open, it is not clear to me whether the below delete call to `"alerts/{}/children/{}"` only deletes the association or even the object itself. Your test seems to indicate the later, but I am doubting it because the child alert is created manually and associated with the composite alert, so I expected the deletion to be symmetrical.